### ISSUE TITLE
Emit references from using declarations to the underlying decl [phab:D1883]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2325]
_Created by **shahms** at 2017-08-16 19:57:31_: 

Emit references from using declarations to the underlying decl

Note that we do not model these as declarations in their own right, merely references to the underlying name.
(as in, we do not model

using ns::Foo;

As:

using Foo = ns::Foo;

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-08-16 20:22:00_:

?

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-08-16 20:22:00_:

Possibly just:
```
  if (auto RCC = ExplicitRangeInCurrentContext(RangeForNameOfDeclaration(Decl))) {
    Observer.recordDeclUseLocation(RCC.primary(),
                                   BuildNodeIdForDecl(Decl->getTargetDecl()),
                                   GraphObserver::Claimability::Claimable);
  }
```
and allow the flags_decls test to ignore_dups (which would only be introduced by this change, and are probably redundant from macro bodies).

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-08-16 20:31:24_:

Heh, debugging (dump() omits a newline).

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-08-16 20:31:24_:

That worked! (and only one of the tests failed with that).  Thanks!

**Comment by kythe** [comment]
_**shahms** wrote at 2017-08-16 20:32:41_:

  - Simplify the code somewhat and ignore_dup for flags_decls to fix the (new) failure.

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2017-08-16 20:47:08_

**Comment by kythe** [comment]
_**shahms** closed via commit 42c36e4bff7495ed7a5292a48dc43c1a620d36d0 at 2017-08-16 20:48:29_
